keras_nlp/utils/text_generation.py

```diff
@@ -17,6 +17,59 @@
 import tensorflow as tf
 
 
+def _validate_prompt(prompt):
+    """
+    Validate the prompt and reformat for use.
+
+    Args:
+        prompt: a list or a Tensor, can be 1D or 2D, the initial tokens to
+            append generated tokens.
+
+    Returns:
+        a 2D Tensor, the prompt with shape [batch_size, max_length].
+    """
+    if isinstance(prompt, tf.RaggedTensor):
+        raise ValueError(
+            "RaggedTensor `prompt` is not supported, please "
+            "provide `prompt` as a list or Tensor."
+        )
+    if not isinstance(prompt, tf.Tensor):
+        prompt = tf.convert_to_tensor(prompt)
+    input_is_1d = prompt.shape.rank == 1
+    if input_is_1d:
+        prompt = prompt[tf.newaxis, :]
+    return prompt, input_is_1d
+
+
+def _mask_tokens_after_end_token(
```

Comment:
We don't need to lead with underscore here. We choose what function to "export" in our API by listing the __init__.py. Underscore just needed for private class methods.

Reply:
Removed underscores for both helper functions